### PR TITLE
Tab bar without id

### LINKED ISSUE #20: Why does TabBar need an ID?
I think it's not very intuitive to require the user to supply an ID. Maybe we can just generate a UUID if it's needed? Minor nitpick, but would make the interface cleaner.

### PR DESCRIPTION
**Pull request by errnesto**
generate id for tabBar automatically

this fixes #20 

also update stories to make tab bar usable.
This reveals a bug(?) where we can't use reacts state hook to just store a tag
the workaround is to store the index so we can still compare the selected tags…

### CODE REVIEW

**Comment by errnesto** [comment]
And I converted it to typescipt because

- I need to write the types anyway as PropType but this generates better documentation
- automatically knows which props are required
- it really helps with wrong properties on components

File: src/components/_shared/TabBar/TabBar.tsx [hunk 1]
```diff
@@ -1,36 +1,54 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, { useRef } from 'react'
+import { uniqueId } from 'lodash'
+import * as colors from 'global_styles/colors'
 import styles from './TabBar.module.sass'
 
+function useId(): string {
+  const { current: id } = useRef(uniqueId())
```
Comment (heyarne) [inline src/components/_shared/TabBar/TabBar.tsx:7]: Nice solution!

File: src/components/_shared/TabBar/TabBar.tsx [hunk 2]
```diff
@@ -1,36 +1,54 @@
...
+  return id
+}
+
+interface Props<T> {
```
Comment (heyarne) [inline src/components/_shared/TabBar/TabBar.tsx:11]: Maybe I don't quite get it, but a `Tab` is not actually a generic, or is it? In the storybook you describe it as always having `{ value: Number, display: String }`, so why not define it as a type?

File: src/components/_shared/TabBar/TabBar.tsx [hunk 3]
```diff
@@ -1,36 +1,54 @@
...
-    id,
-    tabs,
-    title,
-    selectedTab,
-    format,
-    color,
-    onChange,
-    className
-  } = props
+function TabBar<T>({
```
Comment (heyarne) [inline src/components/_shared/TabBar/TabBar.tsx:30]: Same thing here. The `TabBar` always expects the same `Tab`, not some kind of generic `T` that could look like anything.

File: src/components/_shared/TabBar/TabBar.stories.tsx [hunk 4]
```diff
@@ -10,45 +10,50 @@ export default {
...
       format={(tab) => tab.display}
-      color={(tab) => (tab === selectedTab ? tab.color : null)}
-      onChange={action('changed')}
+      color={(tab) => (tab === selectedTab ? tab.color : colors.darkGrey)}
+      onChange={(tab) => {
+        // for some reason setState will make a copy or something so
+        // so selectedTab === tab won't work anymore inside the component
+        // we can't just compare tabs but have to use the index
+        const newIndex = tabs.findIndex((t) => t === tab)
+        setSelectedTabIndex(newIndex)
```
Comment (heyarne) [inline src/components/_shared/TabBar/TabBar.stories.tsx:49]: I first misunderstood the issue: I thought you had to change the actual `TabBar` implementation so it worked with the storybook. The fact that you didn't means that we're probably using a good abstraction that's not leaking. :)

**Pull request merged**